software/base/src/main/java/org/apache/brooklyn/entity/software/base/InboundPortsUtils.java

```diff
@@ -98,4 +107,21 @@
         log.debug("getRequiredOpenPorts detected default {} for {}", ports, entity);
         return ports;
     }
+
+    /**
+     * Checks if a configkey can be managed as a config key {@link PortRange}. The method checks if
+     * the config type or the entity config key value are assignable from the {@link PortRange}.
+     * Moreover the method retrieves true if the config key name matches with pattern.
+     *
+     * @param configuredConfigKeys the configured entity config keys
+     * @param configKey            the configkey that could be assigned as a {@link org.apache.brooklyn.api.location.PortRange}
+     * @param portsPattern         pattern to recognized Config key {@link org.apache.brooklyn.api.location.PortRange} of the name   @return if the config key is recognized as a config key {@link PortRange}
+     */
+    private static boolean isAssignableFromPortConfigKey(Map<ConfigKey<?>, ?> configuredConfigKeys, ConfigKey<?> configKey, Pattern portsPattern) {
```

Comment:
I like this.
Just one thing: The method name only expresses half of the check (the "assignable from" part), not the regex match. I'd recommend keeping it simple and reinstating the regex check out of this method's responsibility (i.e., in the `if` above).